src/index.js

```diff
@@ -39,8 +41,7 @@ exports.unmarshalPublicKey = (buf) => {
 exports.marshalPublicKey = (key, type) => {
   type = (type || 'rsa').toLowerCase()
 
-  // for now only rsa is supported
-  if (type !== 'rsa') {
+  if (type !== 'rsa' && type !== 'ed25519') {
```

Comment:
Now that we have more than on key type (:tada:) could you add a constant at the top `KEY_TYPES`, and check if the type is included in that?